src/mbedtls.c

```diff
@@ -1292,7 +1292,7 @@ _libssh2_mbedtls_ecdsa_new_private(libssh2_ecdsa_ctx **ctx,
     unsigned char *data;
     size_t data_len;
 
-#if MBEDTLS_VERSION_NUMBER >= 0x03060000 && \
+#if MBEDTLS_VERSION_NUMBER >= 0x04000000 && \
     defined(_LIBSSH2_DISABLE_MBEDTLS36_PK_LOAD_FILE)
 
     /* FIXME: implement this functionality via a public API */
```

Comment:
Wouldn't it be better to replace this hack with the FIXME comment, and let it fail when mbedTLS actually decides to remove this function completely?

If so, after a rebase on master, `_LIBSSH2_DISABLE_MBEDTLS36_PK_LOAD_FILE` references need to be deleted from  `.github/workflows/ci.yml`.

Reply:
yes, I agree that it'd better fail to link when this function actually removed: